apps/backoffice/src/components/services/subscriptions-migration/migration-manager.tsx

```diff
@@ -0,0 +1,196 @@
+    fetchData: migrationItemsFetchData
+  } = useFetch<MigrationItemList>();
+  const {
+    data: migrationDelegatesData,
+    loading: migrationDelegatesLoading,
+    fetchData: migrationDelegatesFetchData
+  } = useFetch<MigrationDelegateList>();
+
+  const handleOpenImportModal = () => {
+    setIsImportModalOpen(true);
+  };
+
+  const getMigrationOwnershipClaimsUrl = (delegateId: string) =>
+    `${
+      getConfiguration().API_BACKEND_BASE_URL
+    }/api/services/migrations/ownership-claims/${delegateId}`;
+
+  // direct axios fetch
+  const getDelegateMigrationStatus = async (delegateId: string) => {
+    try {
+      const { data, status } = await axios.get<MigrationData>(
+        getMigrationOwnershipClaimsUrl(delegateId)
+      );
+      return { data, status };
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        return { data: error.message, status: -1 };
+      } else {
+        return { data: JSON.stringify(error), status: -1 };
+      }
+    }
+  };
+
+  // direct axios fetch
+  const postDelegateMigrationOwnershipClaims = async (delegateId: string) => {
+    try {
+      const { status } = await axios.post<void>(
+        getMigrationOwnershipClaimsUrl(delegateId)
+      );
+      return { status };
+    } catch (error) {
+      return { status: -1 };
+    }
+  };
+
+  const fetchDelegateStatusPairs = () => {
+    const pairs: Array<DelegateStatusPair> = [];
+    migrationDelegatesData?.delegates?.forEach(async delegate => {
+      const result = await getDelegateMigrationStatus(delegate.sourceId);
+      if (result.status === 200) {
+        pairs.push({
+          delegate,
+          data: result.data
+        });
+      }
+      setMigrationStatusList([...pairs]);
+    });
+  };
+
+  // claim owneship for selected delegates
+  const claimOwnership = async (delegates: string[]) => {
+    try {
+      setImportInProgress(true);
+      await Promise.all(delegates.map(postDelegateMigrationOwnershipClaims));
+      enqueueSnackbar(
+        buildSnackbarItem({
+          severity: "success",
+          title: t("notifications.success"),
+          message: ""
+        })
+      );
+    } catch (error) {
+      enqueueSnackbar(
+        buildSnackbarItem({
+          severity: "error",
+          title: t("notifications.exceptionError"),
+          message: JSON.stringify(error)
+        })
+      );
+    } finally {
+      setImportInProgress(false);
+      fetchMigrationLatestData();
+    }
+  };
+
+  const fetchMigrationLatestData = () => {
+    migrationItemsFetchData(
+      "getServicesMigrationStatus",
+      {},
+      MigrationItemList,
+      { notify: "errors" }
+    );
+  };
+
+  // for each delegate fetch own migration status
+  useEffect(() => {
+    fetchDelegateStatusPairs();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [migrationDelegatesData]);
+
+  // When import modal is open, fetch delegate list
+  useEffect(() => {
+    if (isImportModalOpen) {
```

Comment:
why don't you put this code into `handleOpenImportModal` instead than `useEffect`?

Reply:
because in this case I chose to use an `effect-based` pattern _(precisely waiting for the modal to change state)_ which can also be triggered by other components.
In this way I rely, with certainty, on the open state of the modal.

Reply:
I'm not sure I get this answer. `isModelOpen` seems to be set only by `onOpenChange`. What's the difference between calling this code directly inside `onOpenChange` handler?

Reply:
(please don't resolve comments without code changes by yourself, otherwise I have to open the accordion for every other comment : -)